test/tests/functional/pbs_power_provisioning_cray.py

```diff
@@ -449,15 +500,536 @@ def test_last_used_time(self):
+        self.server.manager(MGR_CMD_SET, SERVER, {'backfill_depth': '4'})
+        for n in self.server.status(NODE):
+            if 'resources_available.PBScraynid' in n:
+                self.server.manager(MGR_CMD_SET, NODE, {
+                                    "poweroff_eligible": True}, n['id'])
+        a = {"power_on_off_enable": True, 'max_jobs_analyze_limit': '2',
+             'node_idle_limit': '30', 'max_concurrent_nodes': '30'}
+        self.modify_hook_config(attrs=a, id='PBS_power')
+        a = {'freq': 30}
+        self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
+                            sudo=True)
+        a = {'enabled': 'True'}
+        self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
+                            sudo=True)
+        self.server.manager(MGR_CMD_SET, SERVER, {'backfill_depth': '10'})
+        self.logger.info("Waiting for 2 mins to poweroff all the nodes")
+        time.sleep(120)
+        for n in self.server.status(NODE):
+            if 'resources_available.PBScraynid' in n:
+                self.server.expect(NODE, {'state': 'sleep'}, id=n['id'])
+        a = {"node_idle_limit": "1800", 'min_node_down_delay': '30'}
+        self.modify_hook_config(attrs=a, id='PBS_power')
+        j1id = self.submit_job(1000, {'Resource_List.vnode': self.names[0],
+                                      'Resource_List.vntype': 'cray_compute'})
+        j2id = self.submit_job(1000, {'Resource_List.vnode': self.names[1],
+                                      'Resource_List.vntype': 'cray_compute'})
+        j3id = self.submit_job(1000, {'Resource_List.vnode': self.names[2],
+                                      'Resource_List.vntype': 'cray_compute'})
+        j4id = self.submit_job(1000, {'Resource_List.vnode': self.names[3],
+                                      'Resource_List.vntype': 'cray_compute'})
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        self.logger.info(
+            "Waiting for 10 mins to poweron the nodes which are calandered")
+        time.sleep(600)
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        self.server.expect(JOB, {'job_state': 'R'}, id=j1id)
+        self.server.expect(NODE, {'state': 'job-exclusive'}, id=self.names[0])
+        self.server.expect(JOB, {'job_state': 'R'}, id=j2id)
+        self.server.expect(NODE, {'state': 'job-exclusive'}, id=self.names[1])
+        self.server.expect(JOB, {'job_state': 'Q'}, id=j3id)
+        self.server.expect(NODE, {'state': 'sleep'}, id=self.names[2])
+        self.server.expect(JOB, {'job_state': 'Q'}, id=j4id)
+        self.server.expect(NODE, {'state': 'sleep'}, id=self.names[3])
+        self.cleanup_power_on()
+
+    @timeout(1500)
+    def test_preemption_sleep_nodes(self):
+        """
+        Test high priority job preempts a running job rather
+        than waiting for a node to be brought up.
```

Comment:
As a developer, I wouldn't think this test is needed.  The logic to power on/off a node is in a hook.  The preemption logic is in the scheduler.  The scheduler has no clue about nodes being powered on and off, so why would it wait instead of preempting?  It's your test, so it's up to you if you want to keep it.